fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/entity/FabricMinecartComparatorLogicRegistry.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.object.builder.v1.entity;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.vehicle.AbstractMinecartEntity;
+
+public class FabricMinecartComparatorLogicRegistry {
```

Comment:
I'd suggest inlining the `get` and `register` functions into the interface. This will make the overall PR a little bit longer, but the API surface shorter.

Reply:
Don't think it's that much better when you need another class in impl package to hold the map field. Imo the true concerns should be whether we support force overriding (like as is) or throw if a logic is already present.

Reply:
Yeah not sure about inlining.

As for force overriding, I can match what other parts of this module do and just log a message when something is overridden. I don't think throwing is necessary.